src/app/services/qhana-backend.service.ts

```diff
@@ -76,11 +76,14 @@ export interface TimelineSubStepPostData {
 export interface TimelineSubStepApiObject { // FIXME in backend: extends ApiObject
     substepNr: number;
     stepId: number;
-    inputData: string[]; // TODO check this later
+    inputData?: ExperimentDataRef[]; // TODO check this later
     substepId?: string;
     href: string;
     hrefUi?: string;
-    cleared: number; // FIXME fix this once the API correctly returns booleans...
+    cleared: boolean;
+    parameters: string;
```

Comment:
Parameters (and content type) may not be present if cleared is false. Set this to optional if the backend may omit this.